adapter/test/posix/posix_adapter_rs_test.cpp

```diff
@@ -0,0 +1,1184 @@
+            prev_cell_row = current_cell_row;
+            size_t offset = (current_cell_col * cell_stride
+                           + prev_cell_row * cols)
+                          % (info.total_size - info.medium_max);
+            size_t status = lseek(fd, offset, SEEK_SET);
+            REQUIRE(status == offset);
+            size_t request_size = info.medium_min +
+                                (rand_r(&info.rs_seed)
+                                 % info.medium_max);
+            std::string data(request_size, '1');
+            size_t size_read = read(fd, data.data(),
+                                   request_size);
+            REQUIRE(size_read == request_size);
+        }
+        int status = close(fd);
+        REQUIRE(status == 0);
+    }
+    posttest();
+}
+
+TEST_CASE("BatchedUpdateStride2DRSRangeMedium",
+          "[process="+std::to_string(info.comm_size)+"]"
+          "[operation=batched_write]"
+          "[request_size=range-medium][repetition=1024]"
+          "[pattern=stride_2d][file=1]") {
+    pretest();
+    size_t rows = sqrt(info.total_size);
+    size_t cols = rows;
+    REQUIRE(rows * cols == info.total_size);
+    size_t cell_size = 128;
+    size_t cell_stride = rows * cols / cell_size / info.num_iterations;
+    SECTION("write to existing file") {
+        int fd = open(info.existing_file.c_str(), O_RDWR);
+        REQUIRE(fd != -1);
+        size_t prev_cell_col = 0, prev_cell_row = 0;
+        for (size_t i = 0; i < info.num_iterations; ++i) {
+            size_t current_cell_col = (prev_cell_col + cell_stride) % cols;
+            size_t current_cell_row = prev_cell_col + cell_stride > cols ?
+                                    prev_cell_row + 1 : prev_cell_row;
+            prev_cell_row = current_cell_row;
+            size_t offset = (current_cell_col * cell_stride
+                           + prev_cell_row * cols)
+                          % (info.total_size - info.medium_max);
+            size_t status = lseek(fd, offset, SEEK_SET);
+            REQUIRE(status == offset);
+            size_t request_size = info.medium_min +
+                                (rand_r(&info.rs_seed)
+                                 % info.medium_max);
+            std::string data(request_size, '1');
+            size_t size_written = write(fd, data.c_str(),
+                        request_size);
+            REQUIRE(size_written == request_size);
+        }
+        int status = close(fd);
+        REQUIRE(status == 0);
+    }
+    posttest();
+}
+/**
+ * Large RS
+ **/
+
+TEST_CASE("BatchedWriteRSRangeLarge",
+          "[process="+std::to_string(info.comm_size)+"]"
+          "[operation=batched_write]"
+          "[request_size=range-large][repetition=1024]"
+          "[pattern=sequential][file=1]") {
+    pretest();
+    SECTION("write to new file always at the start") {
+        int fd = open(info.new_file.c_str(), O_WRONLY | O_CREAT);
```

Comment:
I'm getting compilation errors in gcc 9.3 everywhere `open` is called with the `O_CREAT` flag and without a `mode` argument. From `man open(2)`:
> The mode argument must be supplied if O_CREAT or O_TMPFILE is specified in flags; if it is not supplied, some arbitrary bytes from the stack will be applied as the file mode.

Here's an example error message:
```
In function ‘int open(const char*, int, ...)’,
    inlined from ‘void ____C_A_T_C_H____T_E_S_T____206()’ at /home/chogan/dev/hermes/adapter/test/posix/posix_adapter_rs_test.cpp:798:22:
/usr/include/x86_64-linux-gnu/bits/fcntl2.h:50:24: error: call to ‘__open_missing_mode’ declared with attribute error: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments
   50 |    __open_missing_mode ();
      |    ~~~~~~~~~~~~~~~~~~~~^~

```

Reply:
I m using gcc 8 and so I am not getting these errors. Anyhow, I have added the mode attribute to calls where I use O_CREATE.

Reply:
I'm still getting compilation errors. You also need to add the mode when you use `O_TMPFILE`.

Reply:
Done.

Reply:
There's still one more error. In `posix_adapter_basic_test.cpp:1329` you're passing 4 arguments to `open`.

Reply:
My compiler missed it. Fixed it now.